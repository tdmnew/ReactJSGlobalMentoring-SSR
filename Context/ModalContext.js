import React from "react";

const useModalState = (initialState) => {
    const [isOpen, setIsOpen] = React.useState(initialState.isOpen);
    const [modalProps, setModalProps] = React.useState(initialState.modalProps);

    const setModalState = ({ isOpen, modalProps = {} }) => {
        setIsOpen(isOpen);
        setModalProps(modalProps);
    };

    return [{ isOpen, modalProps }, setModalState];
};

export const ModalStateContext = React.createContext();
export const ModalUpdaterContext = React.createContext();

export const ModalContext = ({ children }) => {
    const [modalOptions, setModalOptions] = useModalState({
        isOpen: false,
        modalProps: {},
    });

    return (
            <ModalUpdaterContext.Provider value={setModalOptions}>
                <ModalStateContext.Provider value={modalOptions}>
                    {children}
                </ModalStateContext.Provider>
            </ModalUpdaterContext.Provider>
    );
};
