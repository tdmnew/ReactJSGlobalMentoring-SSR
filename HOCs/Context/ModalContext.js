import React, { createContext, useState } from 'react';

const useModalState = (initialState) => {
  const [isOpen, setIsOpen] = useState(initialState.isOpen);
  const [modalProps, setModalProps] = useState(initialState.modalProps);

  const setModalState = ({ isOpen, modalProps = {} }) => {
    setIsOpen(isOpen);
    setModalProps(modalProps);
  };

  return [{ isOpen, modalProps }, setModalState];
};

export const ModalStateContext = createContext();
export const ModalUpdaterContext = createContext();

export const ModalContext = ({ children }) => {
  const [modalOptions, setModalOptions] = useModalState({
    isOpen: false,
    modalProps: {},
  });

  return (
    <ModalUpdaterContext.Provider value={setModalOptions}>
      <ModalStateContext.Provider value={modalOptions}>
        {children}
      </ModalStateContext.Provider>
    </ModalUpdaterContext.Provider>
  );
};
