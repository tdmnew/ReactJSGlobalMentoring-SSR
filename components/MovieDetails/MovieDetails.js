import React from "react";

export default function MovieDetails({movie}) {
    const fallbackImage = (e) => {
        e.target.src =
            "https://linnea.com.ar/wp-content/uploads/2018/09/404PosterNotFound.jpg";
    };

    return (
        <>
            <div className="moviedetails">
                <img
                    className="moviedetails__poster"
                    onError={fallbackImage}
                    src={movie?.poster_path}
                />
                <div className="moviedetails text">
                    <div className="text top">
                        <h2 className="text top__title">{movie?.title}</h2>
                        <span className="text top__rating">
                            {movie?.vote_average}
                        </span>
                    </div>
                    <span className="text genre">{movie?.tagline}</span>
                    <div className="text center">
                        <span className="text center__year">
                            {movie?.release_date !== undefined
                                ? movie.release_date.substring(0, 4)
                                : null}
                        </span>
                        <span className="text center__runtime">
                            {movie?.runtime} min
                        </span>
                        <br />
                    </div>
                    <div className="text bottom">
                        <p className="text bottom__overview">
                            {movie?.overview}
                        </p>
                    </div>
                </div>
            </div>
        </>
    );
}
