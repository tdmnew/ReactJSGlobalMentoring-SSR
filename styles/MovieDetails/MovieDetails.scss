.header {
    display: flex;
    flex-direction: column;
    height: 17rem;
    width: 100%;
    color: #ffffff;
    background-image: linear-gradient(
            to bottom,
            rgba(35, 35, 35, 0.9),
            rgba(35, 35, 35, 0.9)
        ),
        url("https://blackgirlnerds.com/wp-content/uploads/2019/01/Screen-Shot-2019-01-11-at-11.01.10-AM.png");

    &__top {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        align-items: center;

        .logo {
            font-size: 20px;
            color: #f65261;
            padding-left: 3rem;

            &--left {
                font-weight: bold;
            }
            &--right {
                font-weight: lighter;
            }
        }

        .btn--search {
            font-weight: bold;
            border-radius: 3px;
            height: 2rem;
            color: #f65261;
            border: none;
            cursor: pointer;
            position: relative;
            right: 3rem;
            padding: 0 1rem;
            background: url("../../Assets/Images/searching.svg") 50% 50%
                no-repeat;
            background-size: 16px 17px;
        }
    }
}

.moviedetails {
    height: 13.5rem;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;

    &__poster {
        padding: 1rem;
        margin-right: 1rem;
        height: 12rem;
    }

    .text {
        display: flex;
        flex-direction: column;
        width: 35rem;

        @media (max-width: 815px) {
            width: 21.5rem;
        }

        @media (max-width: 530px) {
            width: 15rem;
        }

        @media (max-width: 420px) {
            width: 10rem;
        }

        .top {
            display: flex;
            flex-direction: row;

            @media (max-width: 530px) {
                font-size: 12px;
            }

            &__title {
                margin: 0;
                width: fit-content;
                width: -moz-fit-content;
                padding-right: 1rem;
                font-weight: lighter;
            }

            &__rating {
                color: green;
                text-align: center;
                line-height: 30px;
                height: 2rem;
                width: 2rem;
                border: 1px solid grey;
                border-radius: 20px;
            }
        }

        .genre {
            font-weight: lighter;
        }

        .center {
            display: flex;
            flex-direction: row;
            color: #f65261;
            padding: 0.5rem 0;

            @media (max-width: 530px) {
                font-size: 12px;
            }

            &__year {
                padding-right: 1rem;
            }

            &__year,
            &__runtime {
                width: fit-content;
                width: -moz-fit-content;
            }
        }

        .bottom {
            &__overview {
                font-weight: lighter;
                font-size: 14px;
                margin: 0;

                @media (max-width: 815px) {
                    font-size: 12px;
                }

                @media (max-width: 530px) {
                    font-size: 10px;
                }
            }
        }
    }
}
