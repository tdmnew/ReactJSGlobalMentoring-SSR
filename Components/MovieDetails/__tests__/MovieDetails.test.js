import React from "react";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import { ModalStateContext } from "../../../HOCs/Context/ModalContext";
import MovieDetails from "../MovieDetails";

describe("Movie Card", () => {
    const movie = {
        id: 337167,
        title: 'Fifty Shades Freed',
        tagline: "Don't miss the climax",
        vote_average: 6.1,
        vote_count: 1195,
        release_date: '2018-02-07',
        poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
        overview: 'Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.',
        budget: 55000000,
        revenue: 136906000,
        genres: [ 'Drama', 'Romance' ],
        runtime: 106
    }

    const initialState = {
        movies: [],
        selectedMovie: {}
    }

    const mockStore = configureStore();
    const store = mockStore(initialState);

    it("Renders the movie details page", () => {
        const modalOptions = {
            isOpen: false,
        };

        const tree = render(
            <Provider store={store}>
                <ModalStateContext.Provider value={modalOptions}>
                    <MovieDetails movie={movie} />
                </ModalStateContext.Provider>
            </Provider>
        );

        expect(tree).toMatchSnapshot();
    });
});
