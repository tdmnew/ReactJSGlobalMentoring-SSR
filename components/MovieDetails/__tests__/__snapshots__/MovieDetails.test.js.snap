// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Card Renders the movie details page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="moviedetails"
      >
        <img
          class="moviedetails__poster"
        />
        <div
          class="moviedetails text"
        >
          <div
            class="text top"
          >
            <h2
              class="text top__title"
            />
            <span
              class="text top__rating"
            />
          </div>
          <span
            class="text genre"
          />
          <div
            class="text center"
          >
            <span
              class="text center__year"
            />
            <span
              class="text center__runtime"
            >
               min
            </span>
            <br />
          </div>
          <div
            class="text bottom"
          >
            <p
              class="text bottom__overview"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="moviedetails"
    >
      <img
        class="moviedetails__poster"
      />
      <div
        class="moviedetails text"
      >
        <div
          class="text top"
        >
          <h2
            class="text top__title"
          />
          <span
            class="text top__rating"
          />
        </div>
        <span
          class="text genre"
        />
        <div
          class="text center"
        >
          <span
            class="text center__year"
          />
          <span
            class="text center__runtime"
          >
             min
          </span>
          <br />
        </div>
        <div
          class="text bottom"
        >
          <p
            class="text bottom__overview"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
